### PR TITLE
Allow to launch host processes

### PR DESCRIPTION
**Pull request by letmaik**
This PR allows to trigger host processes from an app endpoint. The app endpoint does not block on it.

- What if the host process sends a response back to CCF (via HTTP) and the relevant app endpoint depends on the transaction that triggered the host process being finished/committed? Does it make sense to investigate an after-commit behaviour?
  - This is ok, assumption is that things are idempotent, so retrying a failed request in this situation would be fine.

- To expose it to JS I temporarily expose the node state as a second argument during `ccf` population. That's a work-around. Related to https://github.com/microsoft/CCF/issues/2448

### CODE REVIEW

**Comment by achamayou** [comment]
This looks good to me, as discussed it would be easier and preferable to use libuv to manage spawning and reaping processes.

The execution is best triggered in the transaction itself rather than hooks, so long as it is idempotent there will be no functional ill-effects.

File: src/host/process_launcher.h [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+#include <chrono>
+#include <queue>
+#include <unordered_map>
+#include <uv.h>
+
+namespace asynchost
+{
+  class ProcessLauncher
+  {
+    static constexpr size_t max_processes = 8;
```
Comment (achamayou) [inline src/host/process_launcher.h:17]: That could really benefit from being a launch-time option.

Comment (letmaik) [reply src/host/process_launcher.h:17]: Sure, but since it's an experimental feature let's wait for feedback first.

Comment (achamayou) [reply src/host/process_launcher.h:17]: I think the best way to know that it's an experimental feature would be for it to be spelled out as such in the changelog ;)

**Review by achamayou — APPROVED** [review]

**Pull request merged**